### PR TITLE
docs: fix links in HACKING.md

### PR DESCRIPTION
**Pull request by dimaqq**
Fix the links to [version.py](ops/version.py) so that they render correctly.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -312,6 +312,5 @@ You can troubleshoot errors on the [Actions Tab](https://github.com/canonical/op
 
 13. Announce the release on [Discourse](https://discourse.charmhub.io/c/framework/42) and [Matrix](https://matrix.to/#/#charmhub-charmdev:ubuntu.com)
 
-14. Open a PR to change [version.py][ops/version.py]'s `version` to the expected
-   next version, with "+dev" appended (for example, if 3.14.1 is the next expected version, use
-   `'3.14.1.dev0'`).
+14. Open a PR to change [version.py](ops/version.py)'s `version` to the expected
+   next version, with ".dev0" appended (for example, if `'3.14.1'` --> `'3.14.1.dev0'`).
```
Comment (benhoyt) [inline HACKING.md:316]: The "if 3.14.1 --> 3.14.1.dev0" reads funny to me. How about, "if 3.14.1, change to 3.14.1.dev0"?

**Review by benhoyt — CHANGES_REQUESTED** [review]

File: HACKING.md (hunk 1 above)
Comment (tonyandrewmeyer) [reply HACKING.md:316]: I don't understand why we're changing the language at all. It's clearer as it was.

**Review by tonyandrewmeyer — CHANGES_REQUESTED** [review]

**Comment by tonyandrewmeyer** [comment]
A description for the PR, even if it's just one sentence, would be good.

File: HACKING.md (hunk 1 above)
Comment (dimaqq) [reply HACKING.md]: reverted the language change, thanks

**Review by tonyandrewmeyer — APPROVED** [review]
Thanks for this - those were my typos, I imagine.

**Comment by dimaqq** [comment]
GitHub indicates that a new review from @benhoyt is necessary

**Review by benhoyt — APPROVED** [review]

**Pull request merged**